### ISSUE TITLE
Inlay hints for rendering applicable marker traits

### ISSUE THREAD

**Issue by Veykril** [rust-lang/rust-analyzer#14055]
Wild idea but I think it would be nice to have something like the following:
```rs
struct Foo {
    field: Bar,
}
```
where we render inlay hint(s) to roughly get: 
```rs
struct Foo /* implements Sized, Send, Sync, ... */{
    field: Bar,
}
```
for the applicable marker traits implemented by the struct.
Bit more tricky is the case where we have generics come into play, though given this is already dreamed up syntax and we can have hover tooltips for them to explain, something like the following might be good:
```rs
struct Foo<T> /* implements Sized, Send if T: Send, Sync if T: Sync, ... */{
    field: T,
}
```
we don't really have the things in place yet to do something like this though

**Comment by flodiebold** [comment]
That might also be more useful for hover than as inlay hints?

**Comment by Veykril** [comment]
it seems useful for both imo

**Comment by zilayo** [comment]
Either hover or inlay hints for traits would be super useful

**Comment by Veykril** [comment]
This is kind of blocked on the new trait solver, as for chalk we can't currently hand it the field types due to perf reasons and thus we can't determine auto trait impls
